cypress/integration/plugins/index-management-dashboards-plugin/snapshots_spec.js

```diff
@@ -53,10 +58,14 @@ describe("Snapshots", () => {
 
       // wait needed here to enable cypress to find "Take snapshot" button.  Timeout
       // cannot be used with cy.createIndex
-      cy.wait(5000);
+      cy.wait(12000);
```

Comment:
@kohinoor98 this PR looks great, I think you are an expert on cypress now 😃

One thing I hope you can help think about and improve is to reduce all the hard coded wait time — basically the long test run time is a tech debt for us

Reply:
@bowenlan-amzn Thanks a lot 😂😂

Yes, I tried to keep it to a minimum; the OSD sometimes takes a lot of time, which is the main reason the test case is flaky. 

Would it be okay if the test case fails sometimes? Another way this issue is handled by other repos - https://github.com/opensearch-project/OpenSearch/pull/10544#issuecomment-1814995523

One of the other reasons the test fails is when the internet connection is unstable. Tested this by throttling the bandwidth to Slow 3g

We could go ahead and reduce the wait to 8500 if the above is ok with you. 

Thanks,
KC

Reply:
I did some reading on cypress introduction guide
Since most of the cypress command support timeout, is the wait here same as increasing timeout to the next command? `cy.get([data-test-subj="takeSnapshotButton"], { timeout: 10000 })`

Reply:
Hi @bowenlan-amzn 

Really sorry I missed this. 

`cy.get` will wait until the object is available, or if the timeout is hit, it will skip and go on. If the object being looked for is available before the timeout, it will move to the following line.
`cy.wait` will wait for the hardcoded amount of time. In this case, it is a lot of time. The new issue #938 should be able to solve it.

We will know more over time!

Is there a blocker for this PR to be merged in?

KC

Reply:
can you move the cy.wait into the its belowing cy.get timeout then, I just don't want to add more time directly onto the cy.wait.

Reply:
@bowenlan-amzn Done!